rasa/core/actions/action.py

```diff
@@ -1009,15 +1006,13 @@ def _matches_mapping_conditions(
     def _verify_mapping_conditions(
         mapping: Dict[Text, Any], tracker: "DialogueStateTracker", slot_name: Text
     ) -> bool:
-        if mapping.get(MAPPING_CONDITIONS) and mapping[MAPPING_TYPE] != str(
-            SlotMappingType.FROM_TRIGGER_INTENT
-        ):
-            if not ActionExtractSlots._matches_mapping_conditions(
+        return bool(
+            not mapping.get(MAPPING_CONDITIONS)
+            or mapping[MAPPING_TYPE] == str(SlotMappingType.FROM_TRIGGER_INTENT)
+            or ActionExtractSlots._matches_mapping_conditions(
                 mapping, tracker, slot_name
-            ):
-                return False
-
-        return True
+            )
+        )
 
```

Comment:
Function `ActionExtractSlots._verify_mapping_conditions` refactored with the following changes:

- Simplify boolean if expression ([boolean-if-exp-identity](https://docs.sourcery.ai/refactorings/boolean-if-exp-identity/))
- Merge nested if conditions ([merge-nested-ifs](https://docs.sourcery.ai/refactorings/merge-nested-ifs/))
- Lift code into else after jump in control flow ([reintroduce-else](https://docs.sourcery.ai/refactorings/reintroduce-else/))
- Replace if statement with if expression ([assign-if-exp](https://docs.sourcery.ai/refactorings/assign-if-exp/))